framework/src/ics/InitialConditionBase.C

```diff
@@ -28,9 +30,15 @@ InitialConditionBase::validParams()
                         "When set to true, a UserObject retrieved "
                         "by this IC will not be executed before the "
                         "this IC");
-
   params.addParamNamesToGroup("ignore_uo_dependency", "Advanced");
 
+  params.addParam<int>("state",
+                       0,
+                       "Specify which state is being set, with 0 being the current state, 1 the "
+                       "old state, 2 the older state, etc..");
```

Comment:
Let's try this:

```suggestion
  MooseEnum stateEnum("CURRENT=0 OLD=1 OLDER=3", "CURRENT", /* allow_out_of_range = */ true);
  params.addParam<int>("state", stateEnum, "Specify state that is being set.");
```

Reply:
Let's just set `allow_out_of_range` to false and not support states older than `OLDER` for now. (so, simply remove the `, /* allow_out_of_range = */ true` above